pkg/apis/serving/k8s_validation.go

```diff
@@ -115,6 +115,19 @@ func validateEnvVar(env corev1.EnvVar) *apis.FieldError {
 		})
 	}
 
+	if env.ValueFrom != nil && env.ValueFrom.FieldRef != nil {
+		switch env.ValueFrom.FieldRef.FieldPath {
+		// Allowed FieldPaths.
+		case "metadata.name", "metadata.namespace", "spec.serviceAccountName":
```

Comment:
`metadata.name` scares me a bit. 

1. As this is going in a `revisionTemplate` I would expect this to be the Revision's name, but it is instead the Pods name. 
2. There is no guarantee that how we name Pods, the Pod lifecycle, requests that get routed to a particular Pod, or how often we re-use old names. Adding this field makes it likely that someone will rely on one of those behaviors to get use out of the field.

Reply:
I use it like the PID of a process to understand what said what. If they are reused or it is the revision in other deployments then that is fine. 

I need an identity. This is the best I have access to, and I use it because it is very specific to this run of the workload.

Reply:
Are you able to generate a UUID at container startup to provide the same PID-type behavior, or is there something specifically more useful about the name?

Reply:
Yeah I can do that. It was helpful for debugging but that cat can be skinned more than one way.